euphoria-core/src/main/java/cz/seznam/euphoria/core/client/operator/ReduceByKey.java

```diff
@@ -352,19 +391,58 @@ public boolean isCombinable() {
     return DAG.of(reduceState);
   }
 
+  static <VALUE> ReduceFunctor<VALUE, VALUE> toReduceFunctor(
+      CombinableReduceFunction<VALUE> reducer1) {
+
+    return new ReduceFunctor<VALUE, VALUE>() {
+      @Override
+      public boolean isCombinable() {
+        return true;
+      }
+
+      @Override
+      public void apply(Iterable<VALUE> elem, Context<VALUE> context) {
+        context.collect(reducer1.apply(elem));
+      }
+    };
+  }
+
+
+  public static class SingleValueContext<T> implements Context<T> {
+
+    T value;
+
+    @Override
+    public void collect(T elem) {
+      value = elem;
+    }
+
+    @Override
+    public Object getWindow() {
+      throw new UnsupportedOperationException(
+          "The window is unknown when applying combinable function");
```

Comment:
The second point is about exposing the `SingleValueContext` - I don't like to expose it on RBK either, but I didn't find a better place to put it and not to reimplement it inside each executor. You are right that it could be put inside the `CombiningReduceState`, but then it would have to be reimplemented inside `euphoria-flink` and `euphoria-spark` appropriately. Maybe I could move this into `cz.seznam.euphoria.core.executor.util` package?